src/v2i-hub/PedestrianPlugin/src/PedestrianPlugin.cpp

```diff
@@ -156,39 +245,43 @@ void PedestrianPlugin::BroadcastPsm(char * psmJson) {  //overloaded
 	PsmMessage psmmessage;
 	PsmEncodedMessage psmENC;
 	tmx::message_container_type container;
-	std::unique_ptr<PsmEncodedMessage> msg;
-
-
-	std::stringstream ss;
-	ss << psmJson;
-
-	container.load<XML>(ss);
-	psmmessage.set_contents(container.get_storage().get_tree());
-
-	const std::string psmString(psmJson);
-	psmENC.encode_j2735_message(psmmessage);
+	std::unique_ptr<PsmEncodedMessage> msg;	
 
+	try
+	{
+		std::stringstream ss;
+		ss << psmJson;
 
-	msg.reset();
-	msg.reset(dynamic_cast<PsmEncodedMessage*>(factory.NewMessage(api::MSGSUBTYPE_PERSONALSAFETYMESSAGE_STRING)));
+		container.load<XML>(ss);
+		psmmessage.set_contents(container.get_storage().get_tree());
 
+		const std::string psmString(psmJson);
 
-	string enc = psmENC.get_encoding();
-	msg->refresh_timestamp();
-	msg->set_payload(psmENC.get_payload_str());
-	msg->set_encoding(enc);
-	msg->set_flags(IvpMsgFlags_RouteDSRC);
-	msg->addDsrcMetadata(172, 0x8002);
-	msg->refresh_timestamp();
+		psmENC.encode_j2735_message(psmmessage);
 
+		msg.reset();
+		msg.reset(dynamic_cast<PsmEncodedMessage*>(factory.NewMessage(api::MSGSUBTYPE_PERSONALSAFETYMESSAGE_STRING)));
 
+		string enc = psmENC.get_encoding();
+		msg->refresh_timestamp();
+		msg->set_payload(psmENC.get_payload_str());
+		msg->set_encoding(enc);
+		msg->set_flags(IvpMsgFlags_RouteDSRC);
+		msg->addDsrcMetadata(172, 0x8002);
+		msg->refresh_timestamp();
 
-	routeable_message *rMsg = dynamic_cast<routeable_message *>(msg.get());
-	BroadcastMessage(*rMsg);
+		routeable_message *rMsg = dynamic_cast<routeable_message *>(msg.get());
+		BroadcastMessage(*rMsg);
 
+		PLOG(logINFO) << " Pedestrian Plugin :: Broadcast PSM:: " << psmENC.get_payload_str() << std::endl;
+	}
+	catch(const std::exception& e)
+	{
+		PLOG(logDEBUG) << "Error: " << e.what() << " broadcasting PSM for xml: " << psmJson << std::endl;
```

Comment:
Any caught exception should probably log at level logWARN, especially one that prevents messages from being broadcast.

Reply:
Done